src/Handler/CurlFactory.php

```diff
@@ -227,12 +227,12 @@ private function applyMethod(EasyHandle $easy, array &$conf)
         }
 
         $method = $easy->request->getMethod();
-        if ($method === 'PUT' || $method === 'POST') {
+        if (($method === 'PUT' || $method === 'POST') && !$easy->request->hasHeader('Content-Length')) {
             // See http://tools.ietf.org/html/rfc7230#section-3.3.2
-            if (!$easy->request->hasHeader('Content-Length')) {
-                $conf[CURLOPT_HTTPHEADER][] = 'Content-Length: 0';
-            }
-        } elseif ($method === 'HEAD') {
+            $conf[CURLOPT_HTTPHEADER][] = 'Content-Length: 0';
+        }
+
+        if ($method === 'HEAD') {
```

Comment:
I am still confused about this: If the method is PUT or POST, I don't want to evaluate this second condition, so elseif is better. But in that case, this needs to be reverted too. Sorry.